# **Introduction**
This model was developed to address a nutritional challenge faced by a petting zoo, which was unable to accurately predict weight changes in penguins based on species, gender, and individual characteristics. By analyzing the correlation between body mass and factors such as bill length, gender, and species, the model provides precise predictions of body mass, enabling the zoo to tailor nutritional plans to the unique needs of each penguin. This approach not only enhances care but also improves the health and well-being of the animals.

In [ ]:
## Load the Libraries

import pandas as pd
import seaborn as sns

## Load the Dataset
penguins = sns.load_dataset("penguins")

## Perform EDA
penguins=penguins[["body_mass_g","bill_length_mm","sex","species"]]
penguins.columns=["body_mass_g","bill_length_mm","gender","species"]
penguins.dropna(inplace=True)
penguins.reset_index(drop=True,inplace=True)
penguins.head()

body_mass_g  bill_length_mm  gender species
0       3750.0            39.1    Male  Adelie
1       3800.0            39.5  Female  Adelie
2       3250.0            40.3  Female  Adelie
3       3450.0            36.7  Female  Adelie
4       3650.0            39.3    Male  Adelie

In [ ]:
## Assign the X & Y variables
penguins_x=penguins[["bill_length_mm","gender","species"]]
penguins_y=penguins[["body_mass_g"]]

## Load & Train the model
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(penguins_x,penguins_y,test_size=0.3,random_state=42)
ols_formula="body_mass_g~bill_length_mm+C(gender)+C(species)"
from statsmodels.formula.api import ols
ols_data=pd.concat([x_train,y_train],axis=1)
OLS=ols(ols_formula,data=ols_data)
model=OLS.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            body_mass_g   R-squared:                       0.850
Model:                            OLS   Adj. R-squared:                  0.847
Method:                 Least Squares   F-statistic:                     322.6
Date:                Sun, 03 Nov 2024   Prob (F-statistic):           1.31e-92
Time:                        06:50:30   Log-Likelihood:                -1671.7
No. Observations:                 233   AIC:                             3353.
Df Residuals:                     228   BIC:                             3371.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
===========================================================================================
                              coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------
Intercept                2032.2111    354.087      5.739      0.000    1334.510    2729.913
C(gender)[T.Male]         528.9508     55.105      9.599      0.000     420.371     637.531
C(species)[T.Chinstrap]  -285.3865    106.339     -2.684      0.008    -494.920     -75.853
C(species)[T.Gentoo]     1081.6246     94.953     11.391      0.000     894.526    1268.723
bill_length_mm             35.5505      9.493      3.745      0.000      16.845      54.256
==============================================================================
Omnibus:                        0.339   Durbin-Watson:                   1.948
Prob(Omnibus):                  0.844   Jarque-Bera (JB):                0.436
Skew:                           0.084   Prob(JB):                        0.804
Kurtosis:                       2.871   Cond. No.                         798.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""


**we can summarize the above table as follows:**
1.   The model expalins 85% of the variance in the dependant varaible Y
2.   since the p value is relatively close to zero, we can say that there is statistical difference between the dpenedant and the independant variables
